### PR TITLE
switch to dune

### PR DESCRIPTION
**Pull request by zapashcanon**
Hi,

I switched the project to dune.

Cheers.

### CODE REVIEW

File: lib_test/dune [hunk 1]
```diff
@@ -0,0 +1,11 @@
+(executable
```
Comment (emillon) [inline lib_test/dune]: You can use a `(test)` stanza which can remove the need for `(alias)` below.

Comment (zapashcanon) [reply lib_test/dune]: Yep', I think I used `dune upgrade` for this one and didn't check properly after. Fixed now.

File: shared-block-ring.opam [hunk 2]
```diff
@@ -10,14 +10,14 @@ tags: [
...
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
 ]
 build-test: [
-  ["jbuilder" "runtest" "-p" name]
+  ["dune" "runtest" "-p" name "-j" jobs]
 ]
 depends: [
-  "jbuilder" {build & >="1.0+beta17"}
+  "dune" {build & >= "1.0"}
```
Comment (avsm) [inline shared-block-ring.opam]:
```suggestion
  "dune"
```

Comment (zapashcanon) [reply shared-block-ring.opam]: Yep', I usually think to remove it, but I used `dune upgrade` and missed it. Thanks !

Comment (avsm) [reply shared-block-ring.opam]: (and thank you for all the other jbuilder -> dune updates as well -- this is much appreciated!)

**Comment by dinosaure** [comment]
Thanks!

**Pull request merged**